### PR TITLE
Make syncing and testing faster

### PR DESCRIPTION
**Pull request by brahmaneya**
1. Use sample data even during tox test.
2. Remove notebook_metadata_filter to allow syncing using jupytext in-browser pairing instead of manually running sync command. 

**Test Plan**
tox. Ran recsys test locally to ensure it is fast, ran recsys sync to make sure it's slow, and tried testing after browser pairing.

### CODE REVIEW

**Comment by brahmaneya** [comment]
> Hm it looks like IS_TEST will still be in the env if we run `tox -e spam — sync`

But it won't have a value in that case right? In the notebooks, we are checking for os.environ.get("IS_TEST") == "true"
not just for the existence of IS_TEST.

**Comment by brahmaneya** [comment]
> > Hm it looks like IS_TEST will still be in the env if we run `tox -e spam — sync`
> 
> But it won't have a value in that case right? In the notebooks, we are checking for os.environ.get("IS_TEST") == "true"
> not just for the existence of IS_TEST.

I've tried running tox -e recsys -- sync and it takes forever, as expected, while tox -e recsys is very fast

File: scripts/build.py [hunk 1]
```diff
@@ -239,13 +239,22 @@ def get_scripts(tutorial_dir: str) -> List[Notebook]:
 
 def check_notebook(notebook: Notebook) -> None:
     assert os.path.exists(notebook.py), f"No file {notebook.py}"
+    os.environ["IS_TEST"] = "true"
```
Comment (henryre) [inline scripts/build.py]: @brahmaneya ah got it, didn't see this line

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -34,7 +34,9 @@ Once you've worked out details with the maintainers, follow these general steps:
 
 1. Make your changes to the source files
     * For notebook-based tutorials, we recommend making changes to the `.py` version
-    then syncing changes with `tox -e my_tutorial_dir -- sync`.
+    then syncing changes with `tox -e my_tutorial_dir -- sync`. Alternatively, if
```
Comment (henryre) [inline CONTRIBUTING.md]: nit (but need to get in the habit): new sentences on new lines in markdown

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -34,7 +34,9 @@ Once you've worked out details with the maintainers, follow these general steps:
...
+    you have already run all the cells in your browser, you can select
+    `File` &rarr; `Jupytext` &rarr; `Pair Notebook with Percept Script` to save the ipynb.
```
Comment (henryre) [inline CONTRIBUTING.md]: nit: typo Percept

unclear. "to save the outputs directly to the notebook version"

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -60,7 +62,9 @@ Once you've worked out details with the maintainers, follow these general steps:
 Also add the command name to `envlist`.
 1. Write your tutorial!
     * For notebook-based tutorials, write your tutorial either as a Python script (e.g. `my_tutorial_dir/my_tutorial.py`) in [Jupytext percent format](https://gist.github.com/mwouts/91f3e1262871cdaa6d35394cd14f9bdc) or a Jupyter notebook.
-        * Run `tox -e my_tutorial_dir -- sync` to generate a notebook version from the Python script version. Run this command to update when changes are made to the tutorial script.
+        * Run `tox -e my_tutorial_dir -- sync` to generate a notebook version from the Python script version
+          (or if you have run all cells, you can select `File` &rarr; `Jupytext` &rarr; `Pair Notebook with Percept Script`).
```
Comment (henryre) [inline CONTRIBUTING.md]: same as above

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -60,7 +62,9 @@ Once you've worked out details with the maintainers, follow these general steps:
...
+          Do this to update the notebook whenever changes are made to the tutorial script.
```
Comment (henryre) [inline CONTRIBUTING.md]: We need to pair every time we make changes?

Comment (brahmaneya) [reply CONTRIBUTING.md]: Updated instructions to unpair after pairing to prevent jupyter from spuriously saving the notebook.

Comment (henryre) [reply CONTRIBUTING.md]: just realized that this requires a jupytext install. and in fact, need to launch the notebook from a sufficient virtualenv. if you rebase, there are some dev instructions. can you add instructions for activating the relevant virtualenv for folks working in notebooks in that section

Comment (brahmaneya) [reply CONTRIBUTING.md]: Added. jupytext is in the requirements, so it'll get installed in the pip install -r requirements.txt.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -28,13 +28,21 @@ We default to using notebook-based tutorials.
...
 ## Dev Setup
 
-The only required dev setup is installing [`tox`](https://tox.readthedocs.io).
-For example, if you use `pip`:
+For dev setup, you  will need to install [`tox`](https://tox.readthedocs.io), and set up a virtualenv with all the requirements.
+For example, if you use `pip`, and want to work on the `spam` tutorial:
 
 ```bash
 python3 -m pip install -U 'tox>=3.13.0,<4.0.0'
+pip3 install --upgrade virtualenv
```
Comment (henryre) [inline CONTRIBUTING.md]: let's use the same `pip` command shown above: `python3 -m pip install`

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -28,13 +28,21 @@ We default to using notebook-based tutorials.
...
+For dev setup, you  will need to install [`tox`](https://tox.readthedocs.io), and set up a virtualenv with all the requirements.
+For example, if you use `pip`, and want to work on the `spam` tutorial:
 
 ```bash
 python3 -m pip install -U 'tox>=3.13.0,<4.0.0'
+python3 -m pip install --upgrade virtualenv
+virtualenv -p python3 .env
+source .env/bin/activate
+
+pip3 install -r requirements.txt
```
Comment (henryre) [inline CONTRIBUTING.md]: here too!!

Comment (henryre) [reply CONTRIBUTING.md]: @brahmaneya

Comment (brahmaneya) [reply CONTRIBUTING.md]: Done. I thought the change was meant for before switching to the venv.

**Pull request merged**